src/checkers/inference/InferenceResult.java

```diff
@@ -18,5 +22,9 @@
      * doesVariableExist(id) is false.
      */
     AnnotationMirror getAnnotation(int varId);
+
+    boolean hasSolution();
+
+    Collection<Constraint> getUnsatisfiableConstraints();
```

Comment:
if `hasSolution()` return `false` and `getUnsatisfiableConstraints()` return `null`, it means the backend doesn't support explanation. It's the client's responsibility to follow this pattern.

Reply:
Having too many APIs related makes `InferenceResult` not orthognal and hard to maintain. `hasSolution` is already related to the remaining three methods in `InferenceResult`. I will use empty map to represent not supporting explanation.